fsh/ig-data/input/pages/index.md

```diff
@@ -97,6 +97,9 @@ for example:
 - App is an HTML5 or JS in-browser app that would expose the secret in user space
 - App is a native app that can only distribute a `client_secret` statically
 
+### Considerations for native apps
+If a native SMART app is unable to guarantee ownership of its `redirect_uri`(s), the app SHALL support Proof Key for Code Exchange (PKCE).  PKCE is a standardized, cross-platform technique for public clients to mitigate the threat of authorization code interception. PKCE is described in [IETF RFC 7636](https://tools.ietf.org/html/rfc7636). SMART requires the S256 code_challenge_method. The "plain" method is not supported. 
```

Comment:
I had added this on the google doc discussion, missed that the PR was here. This language seems a bit fuzzy - how does the app know if they can't guarantee ownership? Can we provide something better here? From some internal discussion:
The point of PKCE is that the O/S and/or browser URI mechanisms can't be trusted, ergo, it would apply to any URL regardless of scheme.
See https://tools.ietf.org/html/rfc8252#section-8.1


"App-claimed "https" scheme redirects are less susceptible to URI
interception due to the presence of the URI authority, but the app is
still a public client; further, the URI is sent using the operating
system's URI dispatch handler with unknown security properties."

The community should follow the BCP guidance here and make it a requirement across the board. There's no way to know whether an app with a cliamed https scheme is native or web, and even then, there'd still be the concern about what kind of browser plugins might be able to monitor URL activity.

Reply:
Thanks! Have updated to match BCP guidelines with "SHALL" for public clients (and "SHOULD" for all clients).